Sources/IO/Geometry/STLWriter/index.js

```diff
@@ -0,0 +1,216 @@
+// Author: Thomas Beznik, with the help of Julien Finet (https://github.com/Kitware/vtk-js/issues/1442)
+// and inspired from Paul Kaplan (https://gist.github.com/paulkaplan/6d5f0ab2c7e8fdc68a61).
+
+import { vec3 } from 'gl-matrix';
+import macro from 'vtk.js/Sources/macro';
+import vtkTriangle from 'vtk.js/Sources/Common/DataModel/Triangle';
+import { FormatTypes } from 'vtk.js/Sources/IO/Geometry/STLWriter/Constants';
+
+const { vtkErrorMacro } = macro;
+
+// ----------------------------------------------------------------------------
+// Global methods
+// ----------------------------------------------------------------------------
+
+function writeFloatBinary(dataview, offset, float) {
+    dataview.setFloat32(offset, float.toPrecision(6), true);
+    return offset + 4;
+}
+
+function writeVectorBinary(dataview, offset, vector) {
+  let off = writeFloatBinary(dataview, offset, vector[0]);
+  off = writeFloatBinary(dataview, off, vector[1]);
+  return writeFloatBinary(dataview, off, vector[2]);
+}
+
+// ----------------------------------------------------------------------------
+// vtkSTLWriter methods
+// ----------------------------------------------------------------------------
+
+function vtkSTLWriter(publicAPI, model) {
+  // Set our className
+  model.classHierarchy.push('vtkSTLWriter');
+
+  function writeBinary(polyData, transform = null) {
```

Comment:
Structure of ascii and binary files is similar .

How about factorizing it with something like the following ?

```
const binaryWriter = () => {
  let offset = 0;
  let dataView = null;
  return {
    init: (polyData) => {
      const buffer = new ArrayBuffer(80 + 4 + (50 * polys.length) / 4);
      dataview = new DataView(buffer);
    },
    writeHeader: (polyData) => {},
    writeTriangle: (v1, v2, v3, dn) => {},
    writeFooter: (polyData) => {},
};
...
writeSTL(polyData, binaryWriter(), transform);
...
function writeSTL(polyData, writer, transform = null) {
  ...
  writer.init(polyData);
  writer.header(polyData);
  for (...) { writer.writeTriangle(...);}
  ...
}
```
Maybe @jourdain has a more elegant way of doing it...

Reply:
That sounds reasonable and seems that they could be used as static methods. If so, they should be exposed as such and used inside the writer class.

That way used could rely on either a writer instance of function to convert a polydata to stl.

```
const stlContent = vtkSTLWriter.convert(polydata, vtkSTLWriter.binaryEncoder);

// or

const writer = vtkSTLWriter.newInstance({ encoding: 'binary' });
writer.setInputdata(polydata);
const stlContent = writer.getOuptuData();
```